song-server/src/main/java/bio/overture/song/server/exceptions/ServerExceptionHandler.java

```diff
@@ -17,118 +17,167 @@
+    }
+  }
+
+  private static void writeResponse(Response response, String content) {
+    try {
+      BufferedWriter out = new BufferedWriter(new OutputStreamWriter(response.getOutputStream()));
+      out.write(content);
+      out.close();
+    } catch (IOException e) {
+      log.error(e.getMessage());
+    }
+  }
 
   @ExceptionHandler(ServerException.class)
-  public ResponseEntity<String> handleServerException(HttpServletRequest request, HttpServletResponse response, ServerException ex){
-    val baseSongError = ex.getSongError();
-    val modifiedSongError = SongError.builder()
-        .debugMessage(baseSongError.getDebugMessage())
-        .httpStatusCode(baseSongError.getHttpStatusCode())
-        .httpStatusName(baseSongError.getHttpStatusName())
-        .message(baseSongError.getMessage())
-        .stackTrace(baseSongError.getStackTrace())
-        .timestamp(baseSongError.getTimestamp())
-        .errorId(baseSongError.getErrorId())
-        .requestUrl(generateRequestUrlWithParams(request))
-        .build();
+  public ResponseEntity<String> handleServerException(HttpServletRequest request, HttpServletResponse response,
+    ServerException ex) {
+    return getServerExceptionResponse(request, ex);
+  }
+
+  private static SongError.SongErrorBuilder withRequestUrl(HttpServletRequest request, SongError e) {
+    return e.toBuilder().requestUrl(generateRequestUrlWithParams(request));
+  }
+
+  private static ResponseEntity<String> getServerExceptionResponse(HttpServletRequest request, ServerException ex) {
+    val modifiedSongError = withRequestUrl(request, ex.getSongError()).build();
     log.error(modifiedSongError.toPrettyJson());
     return modifiedSongError.getResponseEntity();
   }
 
-  @ExceptionHandler(AccessDeniedException.class)
-  public ResponseEntity<String> handleAccessDeniedException(HttpServletRequest request, HttpServletResponse response,
-      AccessDeniedException ex){
-    val rootCause = getRootCause(ex);
-    val error = SongError.builder()
-        .requestUrl(generateRequestUrlWithParams(request))
-        .timestamp(System.currentTimeMillis())
-        .httpStatusCode(UNAUTHORIZED_TOKEN.getHttpStatus().value())
-        .httpStatusName(UNAUTHORIZED_TOKEN.getHttpStatus().name())
-        .errorId(UNAUTHORIZED_TOKEN.getErrorId())
-        .message(ex.getMessage())
-        .debugMessage(format("[ROOT_CAUSE] -> %s: %s", rootCause.getClass().getName(), rootCause.getMessage()))
-        .stackTrace(getFullStackTraceList(ex))
-        .build();
-    log.error(error.toPrettyJson());
-    return error.getResponseEntity();
+  private static String getHttpErrorResponse(HttpServletRequest request, HttpStatusCodeException ex) {
+    val code = ex.getStatusCode();
+    return errorResponse(request, ex, code, code.name(), ex.getMessage());
   }
 
-  @ExceptionHandler(Throwable.class)
-  public ResponseEntity<String> handleThrowable(HttpServletRequest request, HttpServletResponse response, Throwable ex){
```

Comment:
Please add these back in. It is not easy to test these cases, and i believe the song-client depends on the structured output (json) when there is an unauthorized event, which will not even get to the server code that throws a `ServerException`. `SeverException` is thrown inside the service layer. I recall a 401 being returned due to an AccessDeniedException, so thats why that exception handler exists. The `Throwable` exception handler, will nicely wrap any unknown event as an UNKNOWN_ERROR (a 500).